### DISCUSSION TITLE
Security: Handle SNR as position

### DISCUSSION THREAD

**Discussion by mariotti** [meshtastic/Discussions#102] [General]
Hoi All,

INTRO

I found out, that even in a fully private network, I can read SNR information.

The picture below is from a device configured with a PRIMARY channel with a name (not default) and a PSK:
<img width="1314" alt="Meshtastic_Web" src="https://github.com/user-attachments/assets/1df92022-79a8-47ed-af9e-6883f6e3a474">

I was testing 2 nodes communication.

THE ISSUE

You can use the SNR to locate nodes and people. More difficult then the GPS position, but possible.

I can use other device to trace a transmission source,
but in this case we can really trace via SNR specific LoRa and Meshtastic users.

MY SUGGESTION

Handle the SNR information in a similar way as the GPS position.

 - If you share GPS, you share also SNR
 - if you do not share GPS, do not share SNR

ALTERNATIVE

Instead to show numbers just show a 2 or 3 level information: good / (ok) / bad.

The LoRa "locality" is intrinsic, so definitely a node is around you,
a 2 level signal information might prevent from seeking a single node.

FINAL NOTES

What we need to avoid is that I can use "MY" device as a location finder.

There is plenty of other devices that I can use to track an RF source,
but I think it is important that we do not allow explicit tracking.

QUESTIONS

What do you think?

Is it redundant?

The effort does not pay for the actual result?

A+
F

**Comment by RichieFrame** [comment]
I am like, 99% sure that value is provided by the receiving radio hardware and is not 'sent' by the sending radio. It is also relative to the noise floor, a transmission from 100m away could have a 6db then drop to a -6db SNR if there is a 900MHz phone call taking place in the vicinity.

**Comment by garthvh** [comment]
Not sent over the mesh at all.

**Comment by mariotti** [comment]
@garthvh @RichieFrame 

Hoi, I anticipate that I do not need to re-open the issue.
I am also sorry as my RF terminology if very poor :(

As you commented, I'd like to explain better the issue to you.

The SNR is of course not sent by nodes, but it is read by a node as the result of on other node transmitting.
The other node might in the same time transmit the GPS/Location information as a payload.

The question was, if we need to show detailed SNR (on the receiving node) for nodes that do not expose the position.

It is "security by obscurity".

In practice I can use the SNR to "triangulate" the position of a lora node, given a direct connection.
The devices have indeed a unique ID based on the MAC address.

The question was in these lines:
While there are many ways I can track a MAC on LoRa,
Shall we limit the default settings, in such a way that is not possible,
with a simple, just fleshed device with default, 
to track an other direct node position?

I will try to expose my points better the next time :)

A+
F
